makeflow/src/makeflow.c

```diff
@@ -828,6 +849,31 @@ static void makeflow_node_complete(struct dag *d, struct dag_node *n, struct bat
 				makeflow_log_file_state_change(d, f, DAG_FILE_STATE_COMPLETE);
 		}
 
+		/* store node into archiving directory  */
```

Comment:
That's one or.  Or, you could factor out the job expansion code, so that it exists in only once place..